src/main/java/mate/academy/springbookapp/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbookapp.dto.order.OrderDto;
+import mate.academy.springbookapp.dto.order.PlaceOrderRequestDto;
+import mate.academy.springbookapp.dto.order.UpdateStatusRequestDto;
+import mate.academy.springbookapp.dto.orderitem.OrderItemDto;
+import mate.academy.springbookapp.exception.EntityNotFoundException;
+import mate.academy.springbookapp.mapper.OrderItemMapper;
+import mate.academy.springbookapp.mapper.OrderMapper;
+import mate.academy.springbookapp.model.Order;
+import mate.academy.springbookapp.model.OrderItem;
+import mate.academy.springbookapp.model.ShoppingCart;
+import mate.academy.springbookapp.model.User;
+import mate.academy.springbookapp.repository.cartitem.CartItemRepository;
+import mate.academy.springbookapp.repository.order.OrderRepository;
+import mate.academy.springbookapp.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.springbookapp.service.order.OrderService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public List<OrderDto> getAllOrders(Long userId) {
+        return orderRepository.findAllByUserId(userId).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemDto> getAllItemsFromOrder(Long userId, Long orderId) {
+        Order modelOrder = orderRepository.findByIdWithItems(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id: " + orderId));
+        if (!modelOrder.getUser().getId().equals(userId)) {
+            return null;
+        }
+        return modelOrder.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto getItemFromOrder(Long userId, Long orderId, Long itemId) {
+        Order modelOrder = orderRepository.findByIdWithItems(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id: " + orderId));
+        if (!modelOrder.getUser().getId().equals(userId)) {
+            return null;
+        }
+        OrderItem modelItem = modelOrder.getOrderItems().stream()
+                .filter(oi -> oi.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Can't find item with id: " + itemId));
+        return orderItemMapper.toDto(modelItem);
+    }
+
+    @Override
+    public OrderDto placeOrder(User user, PlaceOrderRequestDto requestDto) {
```

Comment:
How about cleaning up this method a bit? First - prepare data, then create new `Order` object and set values. Additionally, you might want to try out Lombok's `@Builder`.